### PR TITLE
docs: Add docs for new entity based metric

### PR DESCRIPTION
**Pull request by sky-2002**
Recently I have added a new metric in #667.
This PR adds documentation for that metric.

### CODE REVIEW

**Comment by shahules786** [comment]
Hey bro @sky-2002 thanks a lot. Like previous, you're blazing fast from idea -> PR. 
I have left few suggestions if you could add that too, we can merge this PR

File: docs/concepts/metrics/context_entities_recall.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+```{hint}
+**Ground truth**: The Taj Mahal is an ivory-white marble mausoleum on the right bank of the river Yamuna in the Indian city of Agra. It was commissioned in 1631 by the Mughal emperor Shah Jahan to house the tomb of his favorite wife, Mumtaz Mahal.
+
+**High entity recall context**: The Taj Mahal is a symbol of love and architectural marvel located in Agra, India. It was built by the Mughal emperor Shah Jahan in memory of his beloved wife, Mumtaz Mahal. The structure is renowned for its intricate marble work and beautiful gardens surrounding it.
+
+**Low entity recall context**: The Taj Mahal is an iconic monument in India. It is a UNESCO World Heritage Site and attracts millions of visitors annually. The intricate carvings and stunning architecture make it a must-visit destination.
+
+````
+
```
Comment (shahules786) [inline docs/concepts/metrics/context_entities_recall.md:20]: can you add a "How was this calculated" section just like [here](https://github.com/explodinggradients/ragas/blob/main/docs/concepts/metrics/faithfulness.md). Helps non-ML users to understand the metric better.

**Review by shahules786 — CHANGES_REQUESTED** [review]

File: docs/concepts/metrics/context_entities_recall.md (hunk 1 above)
Comment (sky-2002) [reply docs/concepts/metrics/context_entities_recall.md:20]: Done. Thanks.

**Review by shahules786 — APPROVED** [review]

File: docs/concepts/metrics/context_entities_recall.md (hunk 1 above)
Comment (shahules786) [reply docs/concepts/metrics/context_entities_recall.md:20]: great, thank you.

**Pull request merged**